pre_commit_hooks/check_links.py

```diff
@@ -32,29 +35,99 @@ def _parse_args(argv: Optional[List[str]] = None) -> argparse.Namespace:
         nargs="+",
         help="One or more paths of files to check.",
     )
+    parser.add_argument(
+        "--anchors-only",
+        action="store_true",
+        help="Set to only validate intra-document anchors, ignoring "
+        " cross-document links.",
+    )
     return parser.parse_args(args=argv)
 
 
-def _check_links(path: str) -> bool:
-    """Updates the table of contents for a file."""
-    with open(path) as f:
-        contents = f.read()
-    headers, links = markdown_links.get_links(contents)
+def _print_error(path: str, link: markdown_links.Link, message: str) -> None:
+    """Prints a link error."""
+    print(
+        f"{path}:{link.line_number}: "
+        f"[{link.label}]({link.destination}): "
+        f"{message}"
+    )
+
+
+class LinkCache(object):
+    """Caches links for a file so that checks don't repeatedly parse files."""
+
+    def __init__(self) -> None:
+        self._cache: Dict[Path, Tuple[Set[str], List[markdown_links.Link]]] = {}
 
-    anchors = set([header.anchor for header in headers])
+    def get(self, path: Path) -> Tuple[Set[str], List[markdown_links.Link]]:
+        assert path.is_absolute(), path
+        if path not in self._cache:
+            with open(path) as f:
+                contents = f.read()
+            headers, links = markdown_links.get_links(contents)
+            anchors = set([header.anchor for header in headers])
+            self._cache[path] = (anchors, links)
+        return self._cache[path]
+
+
+def _check_links(
+    link_cache: LinkCache, repo_root: Path, path: str, anchors_only: bool
+) -> bool:
+    """Updates the table of contents for a file."""
```

Comment:
I think this is old and I just missed it -- but this docstring seems to be stale?

Reply:
Updating the comment